app/controllers/users/omniauth_callbacks_controller.rb

```diff
@@ -40,17 +40,28 @@ def github
   end
 
   def google_oauth2
+    google_auto_register = Errbit::Config.google_auto_register
+    google_domain = Errbit::Config.google_domain
     google_uid = env['omniauth.auth'].uid
     google_email = env['omniauth.auth'].info.email
     google_user = User.where(google_uid: google_uid).first
     google_site_title = Errbit::Config.google_site_title
+
+    if google_domain && !google_email.end_with?("@#{google_domain}")
+      # Fail if the email does not match our domain
+      flash[:error] = "'#{google_email}' is unauthorized. Only #{google_domain} email addresses are allowed."
```

Comment:
Why disclose this information?

Reply:
Good point, should be minimal disclosure. @smerrill you up for fixing this?